test/blackbox/api/dds-pim/PubSubReader.hpp

```diff
@@ -1588,6 +1604,72 @@ class PubSubReader
         }
     }
 
+    void receive_samples(
+            eprosima::fastdds::dds::DataReader* datareader,
+            bool& returnedValue)
+    {
+        eprosima::fastdds::dds::LoanableSequence<type> datas;
+        eprosima::fastdds::dds::SampleInfoSeq infos;
+        returnedValue = true;
+
+        ReturnCode_t success = take_ ?
+                datareader->take(datas, infos) :
+                datareader->read(datas, infos);
+
+        if (!success)
+        {
+            returnedValue = false;
+            return;
+        }
+
+        // Traverse the collection
+        std::unique_lock<std::mutex> lock(mutex_);
+        for (int32_t i = 0; i < datas.length(); ++i)
+        {
+            type& data = datas[i];
+            eprosima::fastdds::dds::SampleInfo& info = infos[i];
+
+            // Validate the sample
+            bool valid_sample = info.valid_data
+                    || datareader->is_sample_valid(&data, &info);
```

Comment:
I think this should be `info.valid_data && datareader->is_sample_valid(&data, &info)`

Field `info.valid_data` will always be `true` for `ALIVE` samples. We need the `&&` in order to check the behavior of `is_sample_valid` for the cases we are interested in.

Reply:
Then I think is better to plainly ignore the `valid_data` member of the info and rely only on the `is_sample_valid` API.